### DISCUSSION TITLE
Switch to openAPI 3.0?

### DISCUSSION THREAD

**Discussion by azhouwd** [containers/podman#17461] [General]
Since openAPI 3.0 has been there for a long time and provides many new features, I'm curious if we have a plan/timeline of switching to 3.0 version?

**Comment by rhatdan** [comment]
Interested in opening a PR to convert this?  As long as this does not cause a breakage in the API, we are not against moving.

**Reply by vrothberg** [reply]
Which APIs are we talking about?  I am a bit lost.

**Comment by Luap99** [comment]
Assuming you talk about the API docs we use https://github.com/go-swagger/go-swagger to generate them based of our code comments. And this tool only supports openapi 2.0, https://github.com/go-swagger/go-swagger/issues/1122

**Reply by azhouwd** [reply]
Thanks, yeah that's exactly what I mean, and I'm just curious whether we are going to switch to other tools that could generate 3.0 spec

**Reply by rhatdan** [reply]
If you knew of other tools and were willing to do the work, I am sure we would consider it.

**Comment by rafalkrupinski** [comment]
Did you mean OpenAPI 3.1?

**Comment by baude** [comment]
looks cool
